File: solar-client/cube.go
package main

import (
	"encoding/binary"
	"github.com/tarm/serial"
	"log"
	"time"
)

func main() {

	c := &serial.Config{Name: "/dev/cu.usbserial-A60205DM", Baud: 9600}

	s, err := serial.OpenPort(c)
	if err != nil {
		log.Fatal(err)
	}

	defer s.Close()

	time.Sleep(3 * time.Second)

	binary.Write(s, binary.LittleEndian, []byte("255,0,0,"))

	s.Flush()

	in := make([]byte, 256)
	n, err := s.Read(in)
	if err != nil {
		log.Fatal(err)
	}

	log.Printf("%q", in[:n])

	for {
		time.Sleep(10 * time.Second)
		log.Println("Waiting...")
	}
}

File: solar-client/cube.py

import serial
import requests
import json
import time

def main():

  ser = serial.Serial('/dev/cu.usbserial-A60205DM', 9600)

  time.sleep(2)

  while True:
    r = requests.get('http://do-jo.co/solar/analysis')

    j = json.loads(r.text)

    count = 0
    r = 0
    g = 0
    b = 0

    for e in j:
      er = e['AverageColor']['R']
      eg = e['AverageColor']['G']
      eb = e['AverageColor']['B']

      esum = er + eg + eb

      if esum >= 50:
        count += 1
        r += er
        g += eg
        b += eb

    r = r / count
    g = g / count
    b = b / count

    s = "%d,%d,%d," % (r, g, b)

    print s, count
    #ser.write(s)

    time.sleep(20)

if __name__ == '__main__':
  main()
